pkg/reactor/download_controller.go

```diff
@@ -111,10 +110,22 @@ func (d *_downloadWorker) Work(ctx context.Context, ctrl *downloadController, ta
 	if task, ok := task.(*DownloadTask); ok {
 		if !ctrl.isDownloaded(task) {
 			if err := d.download(ctx, task); err != nil {
+				// find out any other recorded references that will fail
+				var refs string
+				if tasks, ok := ctrl.downloadedResources[task.Source]; ok {
+					for _, t := range tasks {
+						refs = fmt.Sprintf("%sReference %s from referer %s\n", refs, t.Reference, t.Referer)
+					}
+				}
+				refs = fmt.Sprintf("%sReference %s from referer %s", refs, task.Reference, task.Referer)
+				klog.Warningf("%s : %s\n", refs, err.Error())
+				err = fmt.Errorf("%s : %w", refs, err)
 				return jobs.NewWorkerError(err, 0)
 			}
 			ctrl.setDownloaded(task)
+			return nil
 		}
+		ctrl.setDownloaded(task)
```

Comment:
why do we set it again if ctrl.IsDownloaded returns true

Reply:
(See line 113) To keep track of _all_ referrers that would be affected by a download (failure) and provide sufficient information to trace back the problem to affected documents